### PR TITLE
pkg/operator/sync.go: udpate the namespace for trust configmap

### PR DESCRIPTION
**Pull request by abhinavdahiya**
followup to https://github.com/openshift/installer/pull/2173

/cc @jcpowermac @runcom @danehans

### CODE REVIEW

File: pkg/operator/sync.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
 		return err
 	}
 
-	additionalTrustBundle, err := optr.getCAsFromConfigMap("openshift-config-managed", "user-ca-bundle", "ca-bundle.crt")
+	additionalTrustBundle, err := optr.getCAsFromConfigMap("openshift-config", "user-ca-bundle", "ca-bundle.crt")
```
Comment (bparees) [inline pkg/operator/sync.go]: when is this flow being executed?  is this a one time "during install" flow, in which case it is fine?  or is it a steady state flow, in which case it should be looking at the proxy config resource to determine the name of the configmap.

Comment (runcom) [reply pkg/operator/sync.go]: this is during an MCO sync so it's post-install

Comment (bparees) [reply pkg/operator/sync.go]: yeah, so if you're going to be periodically syncing this information, you need to be looking at the proxy config resource to determine the location(name) of the CA configmap.

The MCO is combining this content w/ the local system trusted from the node?

Comment (runcom) [reply pkg/operator/sync.go]: > The MCO is combining this content w/ the local system trusted from the node?

yes exactly, so you're saying this should be (using the `proxy` status from above):

```
proxy.ProxySpec.TrustedCA.Name
```
?

Comment (runcom) [reply pkg/operator/sync.go]:
```
diff --git a/pkg/operator/sync.go b/pkg/operator/sync.go
index 14b31ad2..608c2514 100644
--- a/pkg/operator/sync.go
+++ b/pkg/operator/sync.go
@@ -155,9 +155,13 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
                return err
        }
 
-       additionalTrustBundle, err := optr.getCAsFromConfigMap("openshift-config-managed", "user-ca-bundle", "ca-bundle.crt")
-       if err != nil && !apierrors.IsNotFound(err) {
-               return err
+       var additionalTrustBundle []byte
+       if proxy != nil && proxy.Spec.TrustedCA.Name != "" {
+               var err error
+               additionalTrustBundle, err := optr.getCAsFromConfigMap("openshift-config-managed", "user-ca-bundle", "ca-bundle.crt")
+               if err != nil && !apierrors.IsNotFound(err) {
+                       return err
+               }
        }
        spec.AdditionalTrustBundle = additionalTrustBundle
 
```

Comment (runcom) [reply pkg/operator/sync.go]:
```diff

diff --git a/pkg/operator/sync.go b/pkg/operator/sync.go
index 14b31ad2..b8148e9c 100644
--- a/pkg/operator/sync.go
+++ b/pkg/operator/sync.go
@@ -155,9 +155,13 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
                return err
        }
 
-       additionalTrustBundle, err := optr.getCAsFromConfigMap("openshift-config-managed", "user-ca-bundle", "ca-bundle.crt")
-       if err != nil && !apierrors.IsNotFound(err) {
-               return err
+       var additionalTrustBundle []byte
+       if proxy != nil && proxy.Spec.TrustedCA.Name != "" {
+               var err error
+               additionalTrustBundle, err = optr.getCAsFromConfigMap("openshift-config-managed", proxy.Spec.TrustedCA.Name, "ca-bundle.crt")
+               if err != nil && !apierrors.IsNotFound(err) {
+                       return err
+               }
        }
        spec.AdditionalTrustBundle = additionalTrustBundle
```

Comment (runcom) [reply pkg/operator/sync.go]: I thought it was always `user-ca-bundle` as defined by the installer

Comment (bparees) [reply pkg/operator/sync.go]: The installer, if it creates it, will create it as that.  The user can create it post install(in the case that they didn't initially need CAs but now they do), or change the name post-install.  In which case they can pick any name they want.

Comment (runcom) [reply pkg/operator/sync.go]: gotcha, thanks for the explaination @bparees - I'll create a PR with the diff above

Comment (runcom) [reply pkg/operator/sync.go]: I guess my confusion from using the proxy.trusteCA cm reference was that additionaltrustBundle could have meant a general way to plug other CAs, not only for proxies

Comment (bparees) [reply pkg/operator/sync.go]: there is almost certainly going to be another turn of the crank on this (post 4.2) where we extend our CA support to be that general way to plug in CAs.  Which may entail another api+configmap name, or another api but sharing a configmap, etc.  But for now the only way we have to identify the CAs the user wants included, is by them specifying it in `proxy.spec.trustbundle`

File: pkg/operator/sync.go [hunk 2]
```diff
@@ -155,11 +155,34 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
...
-	spec.AdditionalTrustBundle = additionalTrustBundle
+	if len(additionalTrustBundle) > 0 {
+		if !certPool.AppendCertsFromPEM(additionalTrustBundle) {
+			return fmt.Errorf("configmap %s/%s doesn't have a valid PEM bundle", "openshift-config", "user-ca-bundle")
+		}
+		trustBundle = append(trustBundle, additionalTrustBundle...)
+	}
+
+	// this is the trusted bundle specific for proxy things and can differ from the generic one above.
+	if proxy.Spec.TrustedCA.Name != "" {
```
Comment (runcom) [inline pkg/operator/sync.go]: should check for nil

File: pkg/operator/sync.go (hunk 1 above)
Comment (runcom) [reply pkg/operator/sync.go]: ack, I think Abhinav modified this patch to account for all of this also

**Comment by runcom** [comment]
/approve

@abhinavdahiya do we need some other ack from someone or testing?

**Comment by abhinavdahiya** [comment]
> /approve
> 
> @abhinavdahiya do we need some other ack from someone or testing?

Not really.. should be fine to merge.

**Pull request merged**